static/js/dashboard-logs-results-grid.js

```diff
@@ -276,14 +276,33 @@ function renderPanelAggsGrid(columnOrder, hits,panelId) {
     $.each(hits, function (key, resMap) {
        newRow.set("id", 0)
        columnOrder.map((colName, index) => {
-           if (resMap.GroupByValues!=null && resMap.GroupByValues[index]!="*" && index< (resMap.GroupByValues).length){
-               newRow.set(colName, resMap.GroupByValues[index])
-           }else{
-               newRow.set(colName, resMap.MeasureVal[colName])
-           }
-       })
+            let ind = -1;
+            if (groupByCols != undefined && groupByCols.length > 0)
+                ind = findColumnIndex(groupByCols, colName)
+            if (resMap.GroupByValues!=null && resMap.GroupByValues[ind]!="*" &&  ind< (resMap.GroupByValues).length && ind != -1){
+                newRow.set(colName, resMap.GroupByValues[ind])
+            }else{
+                // Check if MeasureVal is undefined or null and set it to 0
+                if (resMap.MeasureVal[colName] === undefined || resMap.MeasureVal[colName] === null) {
+                    newRow.set(colName, "0");
+                } else {
+                    newRow.set(colName, resMap.MeasureVal[colName]);
+                }
+            }            
+        })
         segStatsRowData = _.concat(segStatsRowData, Object.fromEntries(newRow));
     })
     aggGridOptions.api.setRowData(segStatsRowData);
     $(`#panel${panelId} .panel-body #panel-loading`).hide();
+}
+
+// Function to find the index of a column in the Map
+function findColumnIndex(columnsMap, columnName) {
```

Comment:
why we repeating code, it was already defined somewhere else, can't be it moved to common.js and use it that way ?